specification/draft-2/README.md

```diff
@@ -0,0 +1,687 @@
+TODO: Intro
+
+ - CWL specifies a way of describing runnable command line tools and workflows.
+ - CWL descriptions are serialized as either YAML or JSON.
+ - A CWL description contains a formal declaration of the inputs and outputs of described process (using Apache Avro schema).
+
+
+# Core specification
+
+CWL specification comes in two parts:
+
+ - Core spec allows description of basic tools and workflows
+ - Extensions add features that enable describing more complex tools and workflows
+
+If using an extension feature, it needs to be explicitly listed in the requirements section
+of the description document. Vendors might also add features incompatible with rest of CWL -
+in this case, they should be explicitly listed as requirements with full URI as value of `class` property.
+
+
+## Command line tools
+
+Basic tool example:
+
+```yaml
+class: CommandLineTool
+label: Grep
+description: Filter lines of text file by regular expression.
+inputs:
+  - id: pattern
+    type: string
+    binding:
+      position: 1
+  - id: in_file
+    type: File
+    streamable: true
+    binding:
+      position: 2
+outputs:
+  - id: out_file
+    type: File
+    streamable: true
+    binding:
+      glob: output.txt
+baseCmd: grep
+stdout: output.txt
+successCodes: [0, 1]
+```
+
+Above is defined a command line tool labeled "grep" with two required inputs
+(a file and a string) and one file output.
+
+Input objects contain a `binding` section which describes how the arguments
+are passed onto the command line. In this case, `pattern` and `in_file` are
+positional arguments and `pattern` is passed first, followed by `in_file`.
+Identifiers for inputs and outputs must be unique across the document.
```

Comment:
Bindings can have same positions (a common case too - default is 0) and the positions do not need to be increments of 1. It simply affects the sort ordering (less means earlier on the command line). In case of same value for position, the ordering should not matter, but they are sorted by input ID.

I'll add a note to the doc.